sys/include/net/sock/util.h

```diff
@@ -22,25 +24,64 @@
  * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
  */
 
-#ifndef NET_SOCK_UTIL_H
-#define NET_SOCK_UTIL_H
-
 #include <stdbool.h>
 #include <stdint.h>
 
-#include "net/sock/udp.h"
-#include "net/sock/tcp.h"
 #include "net/sock/config.h"
 
+#if MODULE_GNRC_SOCK_UDP || MODULE_LWIP_SOCK_UDP || DOXYGEN
+#  include "net/sock/udp.h"
+#endif
+#if MODULE_GNRC_SOCK_TCP || MODULE_LWIP_SOCK_TCP || DOXYGEN
+#  include "net/sock/tcp.h"
+#endif
 #ifdef MODULE_SOCK_DTLS
-#include "net/credman.h"
-#include "net/sock/dtls.h"
+#  include "net/credman.h"
+#  include "net/sock/dtls.h"
+#endif
+
+#if MODULE_GNRC_SOCK_UDP || MODULE_LWIP_SOCK_UDP ||MODULE_GNRC_SOCK_TCP || MODULE_LWIP_SOCK_TCP || DOXYGEN
+#  define HAVE_SOCK_TL_EP   1
```

Comment:
what does this stand for?

Reply:
Both GNRC and lwIP use a single `struct` to hold IP-addr and port for an endpoint, regardless of transport protocol (UDP/TCP) being used.

This adds the flag when that struct had been provided, which is the case when at least one transport protocol is used.

That macro is for internal use; it is used to not copy-paste the same long preprocessor conditional, but to be able to use a simple flag instead.

Reply:
I get that but what does it stand for? What's TL?

Reply:
Transport layer. The flag name matches the type name it indicates availability of; this abbreviation is not introduced here but just consistently used.

Reply:
Oh, yes obviously 😁

Reply:
The Doxygen comment that made this obvious has not been there at the time you asked the question.

But it is good that it is now obvious :)